Resources/public/js/views/services/ez-searchviewservice.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-searchviewservice', function (Y) {
+    "use strict";
+    Y.namespace('eZ');
+
+    Y.eZ.SearchViewService = Y.Base.create('searchViewService', Y.eZ.ViewService, [], {
+        initializer: function () {
+            var app = this.get('app');
+
+            app.on('*:searchRequest', function(e) {
+                app.navigateTo("doSearch", {
+                    searchString: e.searchString,
+                    limit: e.limit,
+                });
+            });
+        },
+
+        /**
+         * Set the searchString and limit from the URL,
+         * If there is a searchString make a fulltext search, stores the results,
+         * and calls `callback` once it's done.
+         *
+         * @method _load
+         * @protected
+         * @param {Function} callback
+         */
+        _load: function (callback) {
+            this.set('searchString', this.get('request').params.searchString);
+            this.set('limit', this.get('request').params.limit ? Number(this.get('request').params.limit) : this.get('loadMoreAddingNumber'));
+            
+            this.search.findLocations({
+                viewName: 'search-' + this.get('searchString'),
+                loadContent: true,
```

Comment:
do we need the Content, the ContentInfo available from each Location should be enough I think